source/scale/high-availability-cluster.rst

```diff
@@ -155,40 +191,89 @@ A sample configuration for NGINX is provided below. It assumes that you have two
+            proxy_buffers 256 16k;
+            proxy_buffer_size 16k;
+            client_body_timeout 60s;
+            send_timeout        300s;
+            lingering_timeout   5s;
+            proxy_connect_timeout   30s;
+            proxy_send_timeout      90s;
+            proxy_read_timeout      90s;
+            proxy_http_version 1.1;
+            proxy_pass http://backend;
       }
 
-      server {
-          server_name mattermost.example.com;
-
-          location ~ /api/v[0-9]+/(users/)?websocket$ {
-                proxy_set_header Upgrade $http_upgrade;
-                proxy_set_header Connection "upgrade";
-                client_max_body_size 50M;
-                proxy_set_header Host $http_host;
-                proxy_set_header X-Real-IP $remote_addr;
-                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-                proxy_set_header X-Forwarded-Proto $scheme;
-                proxy_set_header X-Frame-Options SAMEORIGIN;
-                proxy_buffers 256 16k;
-                proxy_buffer_size 16k;
-                proxy_read_timeout 600s;
-                proxy_http_version 1.1;
-                proxy_pass http://backend;
-          }
-
-          location / {
-                client_max_body_size 50M;
-                proxy_set_header Upgrade $http_upgrade;
-                proxy_set_header Connection "upgrade";
-                proxy_set_header Host $http_host;
-                proxy_set_header X-Real-IP $remote_addr;
-                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-                proxy_set_header X-Forwarded-Proto $scheme;
-                proxy_set_header X-Frame-Options SAMEORIGIN;
-                proxy_pass http://backend;
-          }
+      location ~ /api/v[0-9]+/users/[a-z0-9]+/image$ {
+            proxy_set_header Connection "";
+            client_max_body_size 50M;
+            proxy_set_header Host $http_host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+            proxy_set_header X-Frame-Options SAMEORIGIN;
+            proxy_buffers 256 16k;
+            proxy_buffer_size 16k;
+            client_body_timeout 60s;
+            send_timeout        300s;
+            lingering_timeout   5s;
+            proxy_connect_timeout   30s;
+            proxy_send_timeout      90s;
+            proxy_read_timeout      90s;
+            proxy_http_version 1.1;
+            proxy_pass http://backend;
+            proxy_cache mattermost_cache;
+            proxy_cache_revalidate on;
+            proxy_cache_min_uses 2;
+            proxy_cache_use_stale timeout;
+            proxy_cache_lock on;
+            proxy_ignore_headers Cache-Control Expires;
+            proxy_cache_valid 200 24h;
+      }
+
+      location / {
+            proxy_set_header Connection "";
+            client_max_body_size 50M;
```

Comment:
Can we retain the change we made on https://github.com/mattermost/docs/pull/7021? I also realize that's a different document, not sure if we need both or if we should converge.